### ISSUE TITLE
[Bug]: Runtime Error when using Flair for finetuning NER-model on custom dataset

### ISSUE THREAD

**Issue by SamiNenno** [flairNLP/flair#3304]
### Describe the bug

Hi, 
I am using Flair to train XLM-Roberta on a custom NER dataset (tsv-format below) and receive the error displayed below (screenshot). The code is more or less identical with the one in the tutorial on [Sequence Tagging](https://flairnlp.github.io/docs/tutorial-training/how-to-train-sequence-tagger). The only difference is that I am using the CSVClassificationCorpus in order to load my data. I tried ColumnCorpus as well but it throws errors when loading the data.  I am aware that a similar bug has been [reported before](https://github.com/flairNLP/flair/issues/2801) but the proposed solutions did not work for me. More specifically, setting model_max_length = 512 for the TransformerWordEmbeddings did not change the error.  

text	label
Schartau	B-PER
sagte	O
dem	O
""""	O
Tagesspiegel	B-ORG
""""	O
vom	O
Freitag	O
,	O
Fischer	B-PER

### To Reproduce

```python
from flair.embeddings import TransformerWordEmbeddings
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer
from flair.data import Corpus
from flair.datasets import CSVClassificationCorpus

model_checkpoint = 'xlm-roberta-large'
column_name_map = {0: "text", 1: "label"}
label_type = 'ner'
corpus: Corpus = CSVClassificationCorpus(flair_data_path,
                                         column_name_map,
                                         label_type=label_type,
                                         skip_header=True,
                                         delimiter='\t', 
                                        )
label_dict = corpus.make_label_dictionary(label_type=label_type, add_unk=False)

embeddings = TransformerWordEmbeddings(model=model_checkpoint,
                                       layers="-1",
                                       model_max_length = 512,
                                       subtoken_pooling="first",
                                       fine_tune=True,
                                       use_context=True,
                                       )
tagger = SequenceTagger(hidden_size=256,
                                embeddings=embeddings,
                                tag_dictionary=label_dict,
                                tag_type="ner", #"bio"
                                use_crf=False,
                                use_rnn=False,
                                reproject_embeddings=False,
                        )
trainer = ModelTrainer(tagger, corpus)
trainer.fine_tune(model_path,
                  learning_rate=5.0e-6,
                  mini_batch_size=4,
                  #mini_batch_chunk_size=1, 
                  )
```

### Expected behavior

.

### Screenshots

![Bildschirm­foto 2023-08-22 um 17 09 15](https://github.com/flairNLP/flair/assets/65962113/cd0d4243-8542-4134-b37d-0b5328d44d02)

##### Flair
0.12.2
##### Pytorch
2.0.1+cu117
##### Transformers
4.31.0
#### GPU
True

**Comment by helpmefindaname** [comment]
Hi @SamiNenno the `CSVClassificationCorpus` is for text classification tasks, however you are trying to train a NER model on it. Hence your example currently provides 7 Sentences instead of a single one.
Try using a `ColumnCorpus` so that your example would give 1 Sentences with NER tags instead

**Comment by SamiNenno** [comment]
Hi helpmefindaname,
thank you very much. I did not consider that, it works now 🙌
